src/multiharp/device.rs

```diff
@@ -250,6 +297,61 @@ impl<T: MhlibWrapper> MH160Device<T> {
                 mhlib_wrapper.set_input_channel_enable(channel_id.into(), false)?;
             }
         }
+        if let Some(row) = &config.row_filter {
+            let rows_hw = mhlib_wrapper.get_number_of_rows()?;
+            let rows_hw_usize =
+                usize::try_from(rows_hw).map_err(|_| anyhow!("rows must be non-negative"))?;
+
+            ensure!(
+                row.row_filters.len() <= rows_hw_usize,
+                "too many rows for row filter (given {}, hw rows = {})",
+                row.row_filters.len(),
+                rows_hw
+            );
+
+            for rowidx_usize in 0..rows_hw_usize {
+                if let Some(rf) = row.row_filters.get(rowidx_usize) {
+                    let rowidx: i32 =
+                        i32::try_from(rowidx_usize).map_err(|_| anyhow!("row index overflow"))?;
+
+                    let use_bits: i32 = make_row_mask(&rf.use_channels, rowidx);
+                    let pass_bits: i32 = make_row_mask(&rf.pass_channels, rowidx);
+
+                    mhlib_wrapper.set_row_event_filter(
+                        rowidx,
+                        rf.time_range_ps,
+                        rf.match_count,
+                        rf.invert,
+                        use_bits,
+                        pass_bits,
+                    )?;
+                    mhlib_wrapper.enable_row_event_filter(rowidx, 1)?;
+                } else {
+                    let rowidx: i32 =
+                        i32::try_from(rowidx_usize).map_err(|_| anyhow!("row index overflow"))?;
+                    let _ = mhlib_wrapper.enable_row_event_filter(rowidx, 0);
+                }
+            }
+        }
+
+        if let Some(main) = &config.main_filter {
+            mhlib_wrapper.set_main_event_filter_params(
+                main.time_range_ps,
+                main.match_count,
+                main.invert,
+            )?;
+            mhlib_wrapper.enable_main_event_filter(1)?;
```

Comment:
Basically all of the places we're using raw 1 or 0s need to be replaced with enums or bools to increase legibility.